geneLevel.R

```diff
@@ -41,91 +56,111 @@ IMPACT468_targets = IMPACT468_targets[V5 %like% 'target']
+    if("FACETS_CALL" %in% names(gene_level)) gene_level[, FACETS_CALL := NULL]
 
-  gene_level <- merge(gene_level, FACETS_CALL_table, sort = F, all.x = T)
+    gene_level <- merge(gene_level, FACETS_CALL_table, sort = F, all.x = T)
 
-  gene_level[tcn >= AMP_thresh_tcn, FACETS_CNA := 2]
-  gene_level[tcn >= AMP_thresh_tcn, FACETS_CALL := "AMP"]
+    gene_level[tcn >= AMP_thresh_tcn, FACETS_CNA := 2]
+    gene_level[tcn >= AMP_thresh_tcn, FACETS_CALL := "AMP"]
 
-  output_cols <- c(input_cols, c("FACETS_CNA", "FACETS_CALL"))
-  setcolorder(gene_level, output_cols)
-  setkeyv(gene_level, input_key)
-  gene_level
+    output_cols <- c(input_cols, c("FACETS_CNA", "FACETS_CALL"))
+    setcolorder(gene_level, output_cols)
+    setkeyv(gene_level, input_key)
+    gene_level
 }
 
+
 get_gene_level_calls <- function(cncf_files,
                                  gene_targets,
                                  WGD_threshold = 0.5, ### least value of frac_elev_major_cn for WGD
                                  amp_threshold = 5, ### total copy number greater than this value for an amplification
                                  max_seg_length = 25000000, ### genes in segments longer than this will be treated as diploid
                                  min_cf = 0.6, ### genes in segments with cell fraction less than this will be treated as diploid
-                                 mean_chrom_threshold = 0, ### total copy number also greater than this value multiplied by the chromosome mean for an amplification
-                                 fun.rename = function(filename){filename}){
-
-  ### check version of data.table
-  if(packageVersion("data.table") < "1.9.6"){stop("please update data.table to v1.9.6")}
-
-  ### concatenate input files
-  cncf_txt_list <- lapply(cncf_files, fread)
-  names(cncf_txt_list) <- cncf_files
-#  concat_cncf_txt <- rbindlist(cncf_txt_list, idcol = "filename", fill = T)
-  concat_cncf_txt <- rbindlist(cncf_txt_list, idcol = "filename", fill = T)
-  ### format concat_cncf_txt segment table
-  concat_cncf_txt$chrom <- as.character(concat_cncf_txt$chrom)
-#  concat_cncf_txt[,Tumor_Sample_Barcode := fun.rename(filename)]
-  concat_cncf_txt$Tumor_Sample_Barcode <- as.character(concat_cncf_txt$ID)
-  concat_cncf_txt[, filename := NULL]
-  concat_cncf_txt$chrom <- as.character(concat_cncf_txt$chrom)
-  concat_cncf_txt[chrom == "23", chrom := "X"]
-  setkey(concat_cncf_txt, chrom, loc.start, loc.end)
-
-  if (!("tcn" %in% names(concat_cncf_txt))) {
-    concat_cncf_txt[, c("tcn", "lcn") := list(tcn.em, lcn.em)]
-  }
+                                 mean_chrom_threshold = 0 ### total copy number also greater than this value multiplied by the chromosome mean for an amplification
+                                 ){
+
+    ### check version of data.table
+    if(packageVersion("data.table") < "1.9.6"){stop("please update data.table to v1.9.6")}
+
+
+    ## THE FOLLOWING ASSUMES ONE INPUT FILE, *_hisens.cncf.txt
```

Comment:
Is this true? We normally run this script like below, which loads data from many samples at a time.
```bash
geneLevel.R --filenames /ifs/res/pi/Proj_DMPVal_*/facets/*_hisens.cncf.txt --outfile scna/roslin_gene_level.txt --targetFile IMPACT468 --method scna
```

Reply:
@arichards2564 thanks for removing the comment. I'll mark this bit as resolved.

Reply:
No, sorry, that comment has been removed now. This script will allow for multiple inputs.